src/pvrclient-mythtv.cpp

```diff
@@ -1666,12 +1666,14 @@ PVR_ERROR PVRClientMythTV::AddTimer(const PVR_TIMER &timer)
   XBMC->Log(LOG_DEBUG, "%s: title: %s, start: %ld, end: %ld, chanID: %u", __FUNCTION__, timer.strTitle, timer.startTime, timer.endTime, timer.iClientChannelUid);
   CLockObject lock(m_lock);
   // Check if our timer is a quick recording of live tv
-  // Assumptions: Timer start time = 0, and our live recorder is locked on the same channel.
+  // Assumptions: Our live recorder is locked on the same channel and the recording starts
+  // at the same time as or before (includes 0) the currently in progress program
   // If true then keep recording, setup recorder and let the backend handle the rule.
-  if (timer.startTime == 0 && m_liveStream && m_liveStream->IsPlaying())
+  if (m_liveStream && m_liveStream->IsPlaying())
   {
     Myth::ProgramPtr program = m_liveStream->GetPlayedProgram();
-    if (timer.iClientChannelUid == FindPVRChannelUid(program->channel.chanId))
+    if (timer.iClientChannelUid == FindPVRChannelUid(program->channel.chanId) &&
+        timer.startTime <= program->startTime)
```

Comment:
Short answer: maybe end would work as well, but I haven't tested it.

Long answer: You can submit timers from within live tv playback for shows 
in progress or in the future.
If the current programme being watched matches the selected timer, then we 
need to convert live tv to an instant recording. If the selected programme 
starts after the current live show, then it must be in the future, so we 
need a new recording rule.
If the timer has 0 start time, there isn't an epg entry and we need to 
assume convert live to recording (current code behaviour). Maybe it also 
has zero end time, but I didn't check.
You can group these two conditions using <=.
The case this doesn't cover is a repeating timer (which could be this 
programme, a later one or an earlier one), but I didn't see how to request 
one from within the confluence ui while watching live tv and the original 
code didn't seem to cover this possibility anyway.

Maybe I could look to deal with the repeating timer possibility in a 
selarate pr?

Sent with AquaMail for Android
http://www.aqua-mail.com